### PR TITLE
Adds a cli arg to `run_all_tests.py` for testing a selected extension

### LINKED ISSUE #752: [Proposal] Add ability to run tests per extension
### Proposal

Add a cli arg to run only tests for a selected extension.

Example:
```
isaaclab -t --extension omni.isaac.lab_assets # Only runs tests under omni.isaac.lab_assets
```

### Motivation

When building out a new extension, sometimes a user only wants to specifically test that extension as the full test suite can take long.

### Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)

### Acceptance Criteria

Add the criteria for which this task is considered **done**. If not known at issue creation time, you can add this once the issue is assigned.

- [ ] Tests can be run per extensions, specified via CLI arg to `isaaclab -t`

### PR DESCRIPTION
**Pull request by jsmith-bdai**
# Description

Add a cli argument `--extension` to run only unit tests for a selected extension. As an example:

```bash
isaaclab -t --extension omni.isaac.lab_assets # Only runs tests under omni.isaac.lab_assets
```

Fixes #752 

## Type of change

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

File: tools/run_all_tests.py [hunk 1]
```diff
@@ -126,6 +129,11 @@ def test_all(
                 break
         else:
             raise ValueError(f"Test to skip '{test_to_skip}' not found in tests.")
```
Comment (Mayankm96) [inline tools/run_all_tests.py:131]: Not exactly related but is there a typo here: https://github.com/isaac-sim/IsaacLab/blob/3815c6a554594ce0c4160be58554718d4a4efd9b/tools/run_all_tests.py#L125-L131

File: tools/run_all_tests.py [hunk 2]
```diff
@@ -126,6 +129,11 @@ def test_all(
...
+
+    # Filter tests by extension
+    if extension is not None:
+        all_test_paths = [test_path for test_path in all_test_paths if extension in test_path]
```
Comment (Mayankm96) [inline tools/run_all_tests.py]: This isn't safe. What if I want to only test `omni.isaac.lab`? With the current check, it will still allow the extensions `omni.isaac.lab_assets` and `omni.isaac.lab_tasks` to be included.

File: tools/run_all_tests.py (hunk 1 above)
Comment (jsmith-bdai) [reply tools/run_all_tests.py:131]: Where is the typo?

File: tools/run_all_tests.py (hunk 2 above)
Comment (jsmith-bdai) [reply tools/run_all_tests.py]: Good call! Added a more strict check

File: tools/run_all_tests.py (hunk 1 above)
Comment (Mayankm96) [reply tools/run_all_tests.py:131]: The else case is outside the if?

Comment (Mayankm96) [reply tools/run_all_tests.py:131]: Ah nevermind. Didn't know about the for-else semantics 😅

File: tools/run_all_tests.py [hunk 3]
```diff
@@ -132,7 +132,19 @@ def test_all(
 
     # Filter tests by extension
     if extension is not None:
-        all_test_paths = [test_path for test_path in all_test_paths if extension in test_path]
+        all_tests_in_selected_extension = []
+
+        for test_path in all_test_paths:
+            # Extract extension name from test path
+            extension_name = test_path[test_path.find("extensions") :].split("/")[1]
```
Comment (Mayankm96) [inline tools/run_all_tests.py:139]: If all this is hard-coded to the isaaclab repo, wouldn't it just be simpler to set the test directory itself through `--test_dir`? Wondering now if we even need a special extension case?

Comment (jsmith-bdai) [reply tools/run_all_tests.py:139]: The special extension case enables users to just pass in `omni.isaac.lab_tasks` VS having to specify a whole path which to me is nice. I saw the `test_dir` as a way of specifying a subset (within an extension for example) of tests to run

Comment (Mayankm96) [reply tools/run_all_tests.py:139]: Hm okay. Then the only other thing: Instead of skipping it here as a filtering step, would it make more sense to just add this to the `test_dir` before doing the glob? Unless there is a corner case I'm not seeing, doing this exhaustive for-loop might be slow (depends how many tests we have).

Comment (jsmith-bdai) [reply tools/run_all_tests.py:139]: This for loop would be minimal as far as timing, so I don't think it's worth optimizing given it's easily readable

**Review by Mayankm96 — APPROVED** [review]

**Review by Dhoeller19 — APPROVED** [review]

File: tools/run_all_tests.py (hunk 3 above)
Comment (Mayankm96) [reply tools/run_all_tests.py:139]: Fair enough. LGTM.

**Review by hhansen-bdai — APPROVED** [review]
Nice! LGTM!
